docs/transpile/transpiler-plugins.ipynb

```diff
@@ -0,0 +1,276 @@
+    "\n",
+    "The rest of the page describes how to list available plugins, install new ones, and use them."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## List available plugins and install new ones\n",
+    "\n",
+    "Qiskit already includes some built-in plugins for transpilation. To install more, you can use your Python package manager. For example, you might run `pip install qiskit-toqm` to install the [Qiskit TOQM](https://github.com/qiskit-toqm/qiskit-toqm) routing stage plugin. A number of third-party plugins are part of the [Qiskit ecosystem](https://qiskit.github.io/ecosystem/), and you can find them on [this page](https://qiskit.github.io/ecosystem/#transpiler_plugin).\n",
+    "\n",
+    "### List available transpiler stage plugins\n",
+    "\n",
+    "Use the [list_stage_plugins](/api/qiskit/transpiler_plugins#qiskit.transpiler.preset_passmanagers.plugin.list_stage_plugins) function, passing the name of the stage whose plugins you want to list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['default', 'dense', 'sabre', 'trivial']"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from qiskit.transpiler.preset_passmanagers.plugin import list_stage_plugins\n",
+    "\n",
+    "list_stage_plugins(\"layout\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['basic', 'lookahead', 'none', 'sabre', 'stochastic']"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "list_stage_plugins(\"routing\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " If `qiskit-toqm` were installed, then `toqm` would appear in the list of `routing` plugins."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### List available unitary synthesis plugins\n",
+    "\n",
+    "Use the [unitary_synthesis_plugin_names](api/qiskit/qiskit.transpiler.passes.synthesis.plugin.unitary_synthesis_plugin_names) function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['aqc', 'default', 'sk']"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from qiskit.transpiler.passes.synthesis import unitary_synthesis_plugin_names\n",
+    "\n",
+    "unitary_synthesis_plugin_names()"
```

Comment:
It would be good to link to the documentation of each plugin. See https://github.com/Qiskit/qiskit/issues/11675#issuecomment-1947242111

Reply:
Also, it would be good to use your example in the section "Use a unitary synthesis plugin" below, showing how to use the `default` plugin but setting the decomposer as `TwoQubitControlledUDecomposer(RZXGate)`, passed as a configuration option, presuming such option exists. First we need to find the documentation. See https://github.com/Qiskit/qiskit/issues/11675#issuecomment-1947242111.

Reply:
I found the documentation of the default unitary synthesis: https://docs.quantum.ibm.com/api/qiskit/qiskit.transpiler.passes.synthesis.unitary_synthesis.DefaultUnitarySynthesis

From reading that, it looks like there are no supported options.